logstash-core/lib/logstash/util/substitution_variables.rb

```diff
@@ -76,6 +79,19 @@ def replace_placeholders(value)
       end
       replacement.to_s
     end
+
+    # no further action need if substitution didn't happen
+    return placeholder_value unless is_placeholder_found
+
+    # ENV ${var} value may carry single quote or escaped double quote
+    # or single/double quoted entries in array string, needs to be refined
+    refined_value = placeholder_value.gsub(/[\\"\\']/, '')
+    if refined_value.start_with?('[') && refined_value.end_with?(']')
+      string_array = refined_value[1..-2] # removes the covered [] bracket
+      string_array.include?(',') ? string_array.split(',').map(&:strip) : string_array
```

Comment:
We can be a little more resilient here to whitespace and simplify the code with something like:


```suggestion
    refined_value = placeholder_value.strip.gsub(/[\\"\\']/, '')
    if refined_value.start_with?('[') && refined_value.end_with?(']')
      # remove square brackets, split by comma and cleanup leading/trailing whitespace
      refined_value[1..-2].split(',').map(&:strip)
```

Reply:
It seems `placeholder_value.strip` breaks the agreement in config mixin: https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/1365#01911409-bfda-404f-a783-62b3574cef38

```

1) LogStash::Config::Mixin environment variable evaluation when an environment variable is not set and a default is given should use the default
--
  | Failure/Error: expect(subject.oneArray).to(be == ["first array value", "foo", "", " ", "foo bar"])
  |  
  | expected: == ["first array value", "foo", "", " ", "foo bar"]
  | got:    ["first array value", "foo", "", "", "foo bar"]
  | # ./logstash-core/spec/logstash/config/mixin_spec.rb:536:in `block in <main>'

```